play-json/shared/src/main/scala/Json.scala

```diff
@@ -301,7 +303,10 @@ object Json extends JsonFacade {
    * @define macroWarning If any missing implicit is discovered, compiler will break with corresponding error.
    * @define macroTypeParam @tparam A the type for which the handler must be materialized
    */
-  final class WithOptions[Opts <: MacroOptions]() extends JsonFacade {
+  final class WithOptions[Opts <: MacroOptions](val config: JsonConfiguration.Aux[Opts]) extends JsonFacade {
```

Comment:
So now `WithOptions` keeps the `JsonConfiguration`, which allows the macros to read it.